IHP/ControllerSupport.hs

```diff
@@ -99,13 +105,23 @@ runActionWithNewContext controller = do
 
     try (initContext @application) >>= \case
         Left (exception :: SomeException) -> do
-            case fromException exception of
+            pure $ Left $ case fromException exception of
```

Comment:
Why is this change needed?

Reply:
I think here, the diff is mostly confusing. What I've done is taken the logic from `runActionWithNewContext` and split it into a `newContextForAction` and `runActionWithContext`, then implementing the old `runActionWithNewContext` in terms of those two new functions. So the old function has the same signature and does the same thing. I just needed separate access to the two steps within `RouterSupport`, because I needed a way to create a `ControllerContext` with `initContext` and all other defaults in mind, apply changes to it (from the router), and then run an action with the updated `ControllerContext`.

While writing this, I am realizing that my explanation in the PR is wrong though: first, `initContext` is executed, then the changes by the `router` are applied, not the other way around. I'll have to think about which order makes more sense.

Reply:
> While writing this, I am realizing that my explanation in the PR is wrong though: first, initContext is executed, then the changes by the router are applied, not the other way around. I'll have to think about which order makes more sense.

As a user I'd naively expect the router context to come first and be accessible from the `initContext`

Reply:
I have already changed it to this behavior below :+1: